sippy-ng/src/component_readiness/RegressedTestsModal.js

```diff
@@ -0,0 +1,207 @@
+    expandEnvironment(environmentVal) +
+    `&component=${safeComponentName}` +
+    `&capability=${capabilityName}` +
+    `&platform=${safePlatform}` +
+    `&upgrade=${safeUpgrade}` +
+    `&arch=${safeArch}` +
+    `&network=${safeNetwork}` +
+    `&variant=${safeVariant}` +
+    `&testName=${safeTestName}`
+
+  return sortQueryParams(retUrl)
+}
+
+export default function RegressedTestsModal(props) {
+  // define table columns
+  const columns = [
+    {
+      field: 'test_name',
+      headerName: 'Test Name',
+      flex: 40,
+      renderCell: (param) => <div className="test-name">{param.value}</div>,
+      sortable: false,
+    },
+    {
+      field: 'test_suite',
+      headerName: 'Test Suite',
+      flex: 15,
+      renderCell: (param) => <div className="test-name">{param.value}</div>,
+      sortable: false,
+    },
+    {
+      field: 'component',
+      headerName: 'Component',
+      flex: 20,
+      renderCell: (param) => <div className="test-name">{param.value}</div>,
+      sortable: false,
+    },
+    {
+      field: 'capability',
+      headerName: 'Capability',
+      flex: 12,
+      renderCell: (param) => <div className="test-name">{param.value}</div>,
+      sortable: false,
+    },
+    {
+      field: 'network',
+      headerName: 'Network',
+      flex: 6,
+      sortable: false,
+    },
+    {
+      field: 'upgrade',
+      headerName: 'Upgrade',
+      flex: 10,
+      sortable: false,
+    },
+    {
+      field: 'arch',
+      headerName: 'Arch',
+      flex: 6,
+      sortable: false,
+    },
+    {
+      field: 'platform',
+      headerName: 'Platform',
+      flex: 6,
+      sortable: false,
+    },
+    {
+      field: 'variant',
+      headerName: 'Variant',
+      flex: 8,
+      sortable: false,
+    },
+    {
+      field: 'status',
+      headerName: 'Status',
+      renderCell: (params) => (
+        <div
+          style={{
+            textAlign: 'center',
+          }}
+          className="status"
+        >
+          <Link
+            to={generateTestReport(
+              params.row.test_id,
+              params.row.platform,
+              params.row.upgrade,
+              params.row.arch,
+              params.row.network,
+              params.row.variant,
+              props.filterVals,
+              params.row.component,
+              params.row.capability,
+              params.row.test_name
+            )}
+          >
+            <CompSeverityIcon status={params.value} />
+          </Link>
+        </div>
+      ),
+      flex: 6,
+    },
+  ]
+  return (
+    <Fragment>
+      <Dialog
+        fullWidth={true}
+        maxWidth="xl"
+        open={props.isOpen}
+        onClose={props.close}
+      >
+        <Grid className="regressed-tests-dialog">
+          <Typography
+            variant="h6"
+            style={{ marginTop: 20, marginBottom: 20, marginLeft: 20 }}
+          >
+            Regressed Tests
+          </Typography>
+          <DataGrid
```

Comment:
Can we add a quick filter to this table?

https://mui.com/x/react-data-grid/filtering/quick-filter/